airbase/airbase.py

```diff
@@ -36,32 +31,20 @@ def __init__(self, connect: bool = True) -> None:
             100%|██████████| 5164/5164 [43:39<00:00,  1.95it/s]
             >>> r.download_metadata("data/metadata.tsv")
             Writing metadata to data/metadata.tsv...
-
-        """
-        self._all_countries: list[str] | None = None
-        self._all_pollutants: dict[str, str] | None = None
-        self._pollutants_per_country: dict[
-            str, list[dict[str, str]]
-        ] | None = None
-
-        if connect:
-            self.connect()
-
-    def connect(self, timeout: float | None = None) -> AirbaseClient:
         """
-        Download the available countries and pollutants for validation.
 
-        :param timeout: Raise ConnectionError if the server takes
-            longer than `timeout` seconds to respond.
+        """All countries available from AirBase"""
+        self.all_countries = DB.countries()
 
-        :return: self
-        """
-        summary = fetch_json(E1A_SUMMARY_URL, timeout=timeout)
-        self._all_countries = util.countries_from_summary(summary)
-        self._all_pollutants = util.pollutants_from_summary(summary)
-        self._pollutants_per_country = util.pollutants_per_country(summary)
+        """All pollutants available from AirBase"""
+        self.all_pollutants = DB.pollutants()
 
-        return self
+        """The pollutants available in each country from AirBase."""
+        self.pollutants_per_country: dict[str, list[dict[str, str]]] = dict()
+        for country, pollutants in DB.pollutants_per_country().items():
+            self.pollutants_per_country[country] = [
+                dict(pl=pl, shortpl=str(id)) for pl, id in pollutants.items()
+            ]
```

Comment:
This is only for backwards compatibility.
The URLs are constructed with an int pollutant id.

Reply:
Alright. I don't expect changing to int will cause any issues - lets switch it to int everywhere (I can do that in a separate PR if it's out of scope here) and then bump to 0.4.

Reply:
I did some of the ground work here. The type annotations were a bit tricky as the `pl` and `shortpl` keys have different types. For that I introduced a `typing.TypedDict`

``` python
class PollutantDict(TypedDict):
    pl: str
    shortpl: int
```

Here is only used as a type annotation. On your PR you could use to construct the dicts
and upgrade it to a `dataclasses.dataclass` or a `typing.NemedTuple`.

Reply:
Looks like it's not available by default in 3.7, which I would still like to maintain support for since it's not EOL yet. https://github.com/python/typeshed/issues/3500#issuecomment-560958608

Reply:
ups, I rebased from master and fixed the import with

``` python
if sys.version_info >= (3, 8):
    from typing import TypedDict
else:
    from typing_extensions import TypedDict
```

Reply:
Nice. Any chance you re-included the commits with the `summary.json` when you rebased? I didn't see whether you had already dropped them or not but it would be nice to get it out of the history since it's a bigger file.